include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -850,33 +840,31 @@ class open_addressing_ref_impl {
    * type and presence of other operator mixins.
    *
    * @tparam Value Input type which is implicitly convertible to 'value_type'
-   * @tparam Predicate Predicate type
    *
    * @param slot Pointer to the slot in memory
    * @param value Element to insert
-   * @param predicate Predicate used to compare slot content against `key`
    *
    * @return Result of this operation, i.e., success/continue/duplicate
    */
-  template <typename Value, typename Predicate>
+  template <typename Value>
   [[nodiscard]] __device__ insert_result attempt_insert(value_type* slot,
-                                                        Value const& value,
-                                                        Predicate const& predicate) noexcept
+                                                        Value const& value) noexcept
   {
     if constexpr (sizeof(value_type) <= 8) {
-      return packed_cas(slot, value, predicate);
+      return packed_cas(slot, value);
     } else {
 #if (_CUDA_ARCH__ < 700)
-      return cas_dependent_write(slot, value, predicate);
+      return cas_dependent_write(slot, value);
 #else
-      return back_to_back_cas(slot, value, predicate);
+      return back_to_back_cas(slot, value);
 #endif
     }
   }
 
-  value_type empty_slot_sentinel_;      ///< Sentinel value indicating an empty slot
-  probing_scheme_type probing_scheme_;  ///< Probing scheme
-  storage_ref_type storage_ref_;        ///< Slot storage ref
+  value_type empty_slot_sentinel_;  ///< Sentinel value indicating an empty slot
```

Comment:
```suggestion
```
This can be removed also since the sentinel is wrapped in `predicate_` already

Reply:
I think the predicate only holds the key sentinel, not the slot sentinel, which can be a pair of `{key_sentinel, value_sentinel}`.

Reply:
Yeah, that's fine for this PR. Shortly, we will add the erased sentinel into this equal wrapper as well thus we are duplicating more and more vars.

Can you please put a TODO reminding potential cleanups for sentinels in both wrapper and container ref?

Reply:
Yes, will do. Good point about the `erased_key_sentinel`. We somehow need to rethink this design once we add support for `erase`, since now there are two possible slot sentinels.

Reply:
> since now there are two possible slot sentinels.

That's what I'm struggling with for erase

Reply:
Addressed in [c390393](https://github.com/NVIDIA/cuCollections/pull/379/commits/c39039304418523d1073c2a67ff3911e49839824)